dspace-server-webapp/src/test/java/org/dspace/app/rest/CollectionRestRepositoryIT.java

```diff
@@ -1793,4 +1793,393 @@ public void findAllWithHiddenMetadataTest() throws Exception {
+                                           .withName("Collection 6")
+                                           .withSubmitterGroup(eperson)
+                                           .build();
+
+        Collection col7 = CollectionBuilder.createCollection(context, child1)
+                                           .withName("Collection 7")
+                                           .withSubmitterGroup(eperson)
+                                           .build();
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+        getClient(tokenAdmin).perform(get("/api/core/collections/search/findSubmitAuthorized")
+                 .param("page", "0")
+                 .param("size", "2"))
+                 .andExpect(status().isOk())
+                 .andExpect(content().contentType(contentType))
+                 .andExpect(jsonPath("$._embedded.collections", Matchers.containsInAnyOrder(
+                            CollectionMatcher.matchProperties(col1.getName(), col1.getID(), col1.getHandle()),
+                            CollectionMatcher.matchProperties(col2.getName(), col2.getID(), col2.getHandle())
+                            )))
+                 .andExpect(jsonPath("$._embedded.collections", Matchers.not(Matchers.anyOf(
+                                  is(CollectionMatcher.matchProperties(col3.getName(), col3.getID(), col3.getHandle())),
+                                  is(CollectionMatcher.matchProperties(col4.getName(), col4.getID(), col4.getHandle())),
+                                  is(CollectionMatcher.matchProperties(col5.getName(), col5.getID(), col5.getHandle())),
+                                  is(CollectionMatcher.matchProperties(col6.getName(), col6.getID(), col6.getHandle())),
+                                  is(CollectionMatcher.matchProperties(col7.getName(), col7.getID(), col7.getHandle()))
+                                  ))))
+                 .andExpect(jsonPath("$.page.size", is(2)))
+                 .andExpect(jsonPath("$.page.totalPages", is(4)))
+                 .andExpect(jsonPath("$.page.number", is(0)))
+                 .andExpect(jsonPath("$.page.totalElements", is(7)));
+
+        getClient(tokenAdmin).perform(get("/api/core/collections/search/findSubmitAuthorized")
+                .param("page", "1")
+                .param("size", "2"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$._embedded.collections", Matchers.containsInAnyOrder(
+                           CollectionMatcher.matchProperties(col3.getName(), col3.getID(), col3.getHandle()),
+                           CollectionMatcher.matchProperties(col4.getName(), col4.getID(), col4.getHandle())
+                           )))
+                .andExpect(jsonPath("$._embedded.collections", Matchers.not(Matchers.anyOf(
```

Comment:
you can use `hasSize(2)` to avoid to check for what should be not present, this would make the test shorter and easier to read. There are other method in this class that can benefit of the same change